src/parquet/arrow/writer.cc

```diff
@@ -548,6 +549,37 @@ Status FileWriter::Impl::TypedWriteBatch<ByteArrayType, ::arrow::BinaryType>(
   return Status::OK();
 }
 
+template <>
+Status FileWriter::Impl::TypedWriteBatch<FLBAType, ::arrow::FixedSizeBinaryType>(
+    ColumnWriter* column_writer, const std::shared_ptr<Array>& array, int64_t num_levels,
+    const int16_t* def_levels, const int16_t* rep_levels) {
+  RETURN_NOT_OK(data_buffer_.Resize(array->length() * sizeof(FLBA)));
+  auto data = static_cast<const FixedSizeBinaryArray*>(array.get());
+  auto buffer_ptr = reinterpret_cast<FLBA*>(data_buffer_.mutable_data());
+
+  auto writer = reinterpret_cast<TypedColumnWriter<FLBAType>*>(column_writer);
+
+  if (writer->descr()->schema_node()->is_required() || (data->null_count() == 0)) {
+    // no nulls, just dump the data
+    for (int64_t i = 0; i < data->length(); i++) {
+      buffer_ptr[i] = FixedLenByteArray(data->GetValue(i));
```

Comment:
Since Arrow is a build dependency already, we might consider adding a `WriteBatch` API that can write directly from Arrow arrays without this extra step